File: Data Structures & Algorithms/duplicate-integer/submission-3.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        /*if we use a hashmap we can lookup values
        ,if they dont exist lets put them in there.
        if they already exist, immediately return true.*/
        HashMap<Integer, Integer> seen = new HashMap<>();

        //for loop will check each number in nums
        for(int i  = 0; i < nums.length; i++){
            if(seen.containsKey(nums[i])){
                return true;
            }
            else{
                seen.put(nums[i], 1);
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        for x in range(len(nums)):
            complement = target - nums[x]
            if complement in seen:
                return [seen[complement], x]
            seen[nums[x]] = x